src/bt_peer_connection.cpp

```diff
@@ -1576,6 +1711,149 @@ namespace {
 		stats_counters().inc_stats_counter(counters::num_outgoing_extended);
 	}
 
+	void bt_peer_connection::write_hash_request(hash_request req)
+	{
+		INVARIANT_CHECK;
+
+		char buf[5 + sha256_hash::size() + 4 * 4];
+		char* ptr = buf;
+		detail::write_uint32(int(sizeof(buf) - 4), ptr);
+		detail::write_uint8(msg_hash_request, ptr);
+
+		auto t = associated_torrent().lock();
+		if (!t) return;
+		auto const& ti = t->torrent_file();
+		auto const& fs = ti.files();
+		auto const root = fs.root(req.file);
+
+		ptr = std::copy(root.begin(), root.end(), ptr);
+
+#if 0
+		if (req.base == hash_request::base_layer::leaves)
+			detail::write_uint32(0, ptr);
+		else
+		{
+			// comput the number of layers between the piece layer
+			// and the leaves
+			int blocks_per_piece = ti.piece_length() / (16 * 1024);
+			int layer_index = 0;
+			while (blocks_per_piece >>= 1) ++layer_index;
+			detail::write_uint32(layer_index, ptr);
+		}
+#endif
```

Comment:
we should decide what to do with this block I think. Is this here as a proposed future improvement? (if so, there should probably be a comment to that effect)

Reply:
This is obsolete, I'll remove it.